presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -581,8 +586,8 @@ protected Object visitInPredicate(InPredicate node, Object context)
                     set = FastutilSetHelper.toFastutilHashSet(
                             objectSet,
                             type,
-                            metadata.getScalarFunctionInvoker(metadata.resolveOperator(HASH_CODE, ImmutableList.of(type)), Optional.empty()).getMethodHandle(),
-                            metadata.getScalarFunctionInvoker(metadata.resolveOperator(EQUAL, ImmutableList.of(type, type)), Optional.empty()).getMethodHandle());
+                            metadata.getScalarFunctionInvoker(metadata.resolveOperator(HASH_CODE, ImmutableList.of(type)), simpleConvention(FAIL_ON_NULL, NEVER_NULL)).getMethodHandle(),
+                            metadata.getScalarFunctionInvoker(metadata.resolveOperator(EQUAL, ImmutableList.of(type, type)), simpleConvention(NULLABLE_RETURN, NEVER_NULL, NEVER_NULL)).getMethodHandle());
```

Comment:
Why are we getting an EQUALs operator that allows nullable return? That doesn't seem expected for equals as an operator?

Reply:
i guess `ARRAY[NULL] = ARRAY[1]` should return `NULL`

Reply:
Yep that is why